tools/setup_helpers/dist_check.py

```diff
@@ -58,10 +71,46 @@ def should_build_ib():
         #   mlx5_2
         #   mlx5_1
         #   mlx5_0
-        return True
+        ib_util_found = True
     except Exception:
         # We just take all the exceptions here without affecting the build
-        return False
+        ib_util_found = False
+
+    lib_paths = list(filter(bool, [
+        "/usr/lib/",
+        "/usr/lib/x86_64-linux-gnu/",
+        "/usr/lib/powerpc64le-linux-gnu/",
+        "/usr/lib/aarch64-linux-gnu/",
+    ] + gather_paths([
+        "LIBRARY_PATH",
+    ]) + gather_paths([
+        "LD_LIBRARY_PATH",
+    ])))
```

Comment:
Three's a charm for deduplicating (this is copypasted in nccl and cudnn)

Reply:
gather_paths is now in a common place :)